taskvine/src/manager/vine_manager.c

```diff
@@ -465,13 +466,25 @@ on its own port to receive get requests from other workers.
 static int handle_transfer_address(struct vine_manager *q, struct vine_worker_info *w, const char *line)
 {
 	int dummy_port;
-	if (sscanf(line, "transfer-address %s %d", w->transfer_addr, &w->transfer_port)) {
-		w->transfer_port_active = 1;
-		link_address_remote(w->link, w->transfer_addr, &dummy_port);
-		return VINE_MSG_PROCESSED;
-	} else {
+	int explicit;
```

Comment:
Hmm, I'm concerned that this is tripping over some existing foot guns, and then also creating some new ones.  I had forgotten that the old worker code didn't really send the right worker address anyway (since normally it would listen on any interface.). And so now this overloads the `transfer-address` method with two or three strange subclasses.

I suggest that you break it apart doing something like this:
- When the worker listens normally on all interfaces on a specific port, have it send `transfer-port` with just the port number, and the manager should assume `link_address_remote` as before.
- When the worker sends a (potentially fake) host and port, then let's use a separate message `transfer-hostport` for that case.

Reply:
And then in both cases it constructs `w->transfer_url` as appropriate.